show-fixed-kernel-cves.py

```diff
@@ -1,58 +1,131 @@
-# also what can be found under https://www.linuxkernelcves.com/streams/5.15
-
-import json
-from packaging import version
+"""
+Helper to show which Linux kernel CVEs got fixed in a given version
+
+```
+virtualenv venv
+source venv/bin/activate
+pip install feedparser
+python show-fixed-kernel-cves.py --from_version 6.6.32 --to_version 6.6.44
+```
+"""
 from optparse import OptionParser
-import urllib.request
-
-def print_fixed_linux_cves(from_version_str, to_version_str):
-  stream_data_url = "https://raw.githubusercontent.com/nluedtke/linux_kernel_cves/master/data/stream_data.json"
-  payload = urllib.request.urlopen(stream_data_url).read()
-  streams = json.loads(payload)
-  from_version=version.Version(from_version_str)
-  to_version=version.Version(to_version_str)
-  cvelist = []
-  links = []
-  for stream, releases in streams.items():
-    for release, cves in releases.items():
-      if release != "outstanding" and from_version < version.Version(release) <= to_version:
-        cvelist += cves.keys()
-  for cve in sorted(cvelist):
-    links += [f"[{cve}](https://nvd.nist.gov/vuln/detail/{cve})"]
-  print(", ".join(links))
+import subprocess
+
+import feedparser
+
+class Version:
+    """
+    Version implement a simple semver version object.
+    """
+    def __init__(self, v):
+        # Clean the 'v' from the version (e.g 'v6.6.44')
+        if v.startswith("v"):
+            v = v[1:]
+
+        s = v.split(".")
+        if len(s) != 3:
+            self.major = "-1"
+            self.minor = "-1"
+            self.patch = "-1"
+        else:
+            self.major = s[0]
+            self.minor = s[1]
+            self.patch = s[2]
+
+    def __str__(self):
+        return f"{self.major}.{self.minor}.{self.patch}"
+
+    def __eq__(self, v):
+        if isinstance(v, Version):
+            return self.major == v.major \
+                    and self.minor == v.minor \
+                    and self.patch == v.patch
+
+        return False
+
+    def __lt__(self, v):
+        if self.major < v.major:
+            return True
+        if self.major > v.major:
+            return False
+        if self.major == v.major:
+            if self.minor < v.minor:
+                return True
+            if self.minor > v.minor:
+                return False
+            if self.minor == v.minor:
+                if self.patch < v.patch:
+                    return True
+                return False
+
+    def __ge__(self, v):
+        if self == v:
+            return True
+        if self.major > v.major:
+            return True
+        if self.major < v.major:
+            return False
+        if self.major == v.major:
+            if self.minor > v.minor:
+                return True
+            if self.minor < v.minor:
+                return False
+            if self.minor == v.minor:
+                if self.patch > v.patch:
+                    return True
+                return False
```

Comment:
If this function can't be `return v < self`, then this part ought to be
```suggestion
                return self.patch >= v.patch
```

or:
```suggestion
                if self.patch >= v.patch:
                    return True
                return False
```